GameDefinition/game-src/TieKnot.hs

```diff
@@ -62,7 +62,7 @@ tieKnotForAsync options@ServerOptions{ sallClear
   -- This setup ensures the boosting option doesn't affect generating initial
   -- RNG for dungeon, etc., and also, that setting dungeon RNG on commandline
   -- equal to what was generated last time, ensures the same item boost.
-  initialGen <- maybe R.getStdGen return sdungeonRng
+  initialGen <- maybe SM.newSMGen return sdungeonRng
```

Comment:
This is fine. The point is that we take that global state thing only once (if ever), but use many times later on, in particular to print it, so that the game run can be replicated exactly.